src/System.Data.SqlClient/ref/project.json

```diff
@@ -5,7 +5,8 @@
     "System.IO": "4.0.0",
     "System.Runtime": "4.0.0",
     "System.Threading.Tasks": "4.0.0",
-    "System.Xml.ReaderWriter": "4.0.0"
+    "System.Xml.ReaderWriter": "4.0.0",
+	"System.Collections.NonGeneric": "4.0.0",
```

Comment:
@justinvp Thats a good idea. 
I want to take this up as a part of a separate item. I want this PR to take care of two versions of SqlClient contracts targeting different .Net frameworks.
Once this is done I will track your suggestions with an issue.